### PR TITLE
Add support for IPv6 Addresses

### PR DESCRIPTION
**Pull request by belohnung**
The InetSocketAddress deserialization code couldn't handle IPv6 Addresses, also there was a property set which prevented IPv6 Addresses from binding.

### CODE REVIEW

**Comment by TobiasG-DE** [comment]
Is this tested to work with IPv6?

File: src/main/java/dev/waterdog/utils/ProxyConfig.java [hunk 1]
```diff
@@ -85,6 +85,10 @@
     @Comment("If enabled, only players which are authenticated with XBOX Live can join. If disabled, anyone can connect *with any name*")
     private boolean onlineMode = true;
 
+    @Path("enable_ipv6")
+    @Comment("If enabled, the proxy will be able to bind to an Ipv6 Address")
+    private boolean enableIpv6 = true;
```
Comment (Alemiz112) [inline src/main/java/dev/waterdog/utils/ProxyConfig.java:90]: Some networks may not support ipv6, lets disable it be default.

**Review by Alemiz112 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/waterdog/utils/ProxyConfig.java (hunk 1 above)
Comment (TobiasG-DE) [reply src/main/java/dev/waterdog/utils/ProxyConfig.java:90]: Agreed

**Comment by belohnung** [comment]
I tested it and it works just fine, also implemented your proposed change.

**Review by Alemiz112 — APPROVED** [review]
Approved

**Pull request merged**